portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -20,28 +20,51 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.util.ArrayList;
+import java.util.List;
 import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  private ArrayList<String> commentSection = new ArrayList<String>();
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String commentsJson = new Gson().toJson(commentSection);
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);;
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<String> commentSection = new ArrayList<String>();
+    for (Entity entity : results.asIterable()) {
+        String text = (String) entity.getProperty("text");
+
+        String comment = text;
+        commentSection.add(comment);
```

Comment:
Is there a reason for the two separate `String` variables here?

Reply:
Nope, that was a mistake that I saw later. Thank you!